### PR TITLE
kata 6

### PR DESCRIPTION
**Pull request by RTXVader**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -118,18 +118,56 @@ describe("GET /api/articles", () => {
...
+        const { comments } = body;
+        comments.forEach((comment) => { 
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("votes", expect.any(Number));
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+        
+        });
```
Comment (yalicee) [inline __tests__/app.test.js]: you can just combine these all to 1 test. Also if you use a forEach then you know its an array so first test is redundant.

**Review by yalicee — CHANGES_REQUESTED** [review]
Add these tests please:

- Status 200, valid ID, but has no comments responds with an empty array of comments
- Status 400, invalid ID, e.g. string of "not-an-id"
- Status 404, non existent ID, e.g. 0 or 9999

File: __tests__/app.test.js [hunk 2]
```diff
@@ -118,18 +118,80 @@ describe("GET /api/articles", () => {
...
+      });
+  });
+  it('200: valid id but no comments respond with an empty array of comments', () => {
+    return request(app)
+      .get('/api/articles/2/comments')
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+
```
Comment (yalicee) [inline __tests__/app.test.js]: Test for empty array as well

File: modele/app.model.js [hunk 3]
```diff
@@ -8,41 +8,82 @@ exports.gettingTopics = () => {
...
+
+  const query = `
+    SELECT *
+    FROM comments
+    WHERE article_id = $1
+    ORDER BY created_at DESC;
+  `;
+
+  return db.query(articleQuery, [article_id]).then(({ rows }) => {
+    const articleExists = rows[0].exists;
```
Comment (yalicee) [inline modele/app.model.js:77]: What is the .exists? Is it a property on the article object? Do you need it?

**Review by yalicee — APPROVED** [review]
Make minor changes before you merge

File: modele/app.model.js (hunk 3 above)
Comment (RTXVader) [reply modele/app.model.js:77]: I think so its the boolean outcome of the article query to see if the article id belongs to an article

File: __tests__/app.test.js (hunk 2 above)
Comment (RTXVader) [reply __tests__/app.test.js]: is that for the 200: valid id but no comments respond with an empty array of comments test

Comment (yalicee) [reply __tests__/app.test.js]: Ye

Comment (RTXVader) [reply __tests__/app.test.js]: didn't I do it on line 177 with expect(comments).toHaveLength(0)

**Pull request merged**